tools/generate_buildfiles.py

```diff
@@ -68,7 +68,7 @@ def RunHostGn(options):
         '-m',
         'all',
         '-a',
-        'all',
```

Comment:
`ParseArgs` below does not actually declare an option called `arch`. So running `tools/generate_buildfiles.py --arch x64` would just generated an error about unrecognised option.

Reply:
Well, from my experiment, https://github.com/dart-lang/sdk/issues/49982#issue-1376645273, `./tools/generate_idefiles.py --verbose --arch x64` works with modification to this code. Please see full logs there

Reply:
@fzyzcjy I see `./tools/generate_idefiles.py` has its own argument parser. Consider adding the following

```python
    parser.add_argument("-a",
                        "--arch",
                        help="Target architecture for runtime sources.",
                        default="x64")
``` 

to `ParseArgs` so that `generate_buildfiles` could run on its own with `--arch` as well.

Reply:
Thanks, I will do that

Reply:
hi @fzyzcjy still working on this?

Reply:
@mit-mit Ah I forget about this PR. The workaround I mentioned work well for me so I may not have bandwidth to finish this PR now. Since I am working on https://flutter.dev/go/sixty-fps

Reply:
No problem, I'll close this out for now